hudi-client/hudi-flink-client/src/main/java/org/apache/hudi/io/FlinkMergeAndReplaceHandle.java

```diff
@@ -133,6 +135,17 @@ protected void makeOldAndNewFilePaths(String partitionPath, String oldFileName,
     }
   }
 
+  @Override
+  protected HoodieRecord<T> updateOrPrependMetaFields(HoodieRecord<T> record, Schema schema, Schema targetSchema, String fileName, Properties prop) {
+    if (schema.getField(HoodieRecord.RECORD_KEY_METADATA_FIELD) == null) {
```

Comment:
there is a flag in the merge handle to indicate whether the record includes metadata fields.

Reply:
ok, will update